04-Multitasking/context_switch.S

```diff
@@ -1,5 +1,16 @@
 .syntax unified
 
+/* check the situaion and determine the path to return */
+.macro ret_path, scratch
+	mov \scratch, #0xfffffff0
+	cmp lr, \scratch  /* lr - 0xfffffff0 */
+
+	ittt ls /* `ls` condition means unsinged lower or the same */
```

Comment:
Only once. But I think it is more clear and flexible than leave hard code in `activate()`.